lib/Doctrine/ORM/Query/Expr/From.php

```diff
@@ -21,9 +21,9 @@ class From
     protected $indexBy;
 
     /**
-     * @param string $from    The class name.
-     * @param string $alias   The alias of the class.
-     * @param string $indexBy The index for the from.
+     * @param class-string $from    The class name.
```

Comment:
I'm pretty sure that this is not correct. I don't necessarily need to reference an entity by FCQN, so while `$from` _can_ be a `class-string`, it does not have to be.

Reply:
Are `:`-separated entity names supported in ORM 2.13? I'm not sure what the non-FQCN values are.

Reply:
> I'm pretty sure that this is not correct. I don't necessarily need to reference an entity by FCQN, so while `$from` _can_ be a `class-string`, it does not have to be.

Do you have an example ? @derrabus (and especially a non-deprecated example)

Reply:
They are, although the syntax is deprecated. Yet it's still a supported feature. I was also under the impression that we could omit the namespace in certain situations, but that does not seem to be the case here.

Reply:
#9777 is a different story because we deprecate something that has always been wrong. Today's tooling allows us to be more precise, but the documentation has always stated that those two constants were the only valid values. Technically, we could've just thrown an exception instead and it wouldn't have been a BC break. We're being nice by going through that deprecation process.

---

In the case of this PR, we're talking about a well-documented feature that is being sunset. This is something we need to be more careful about because otherwise our static analyzers would consider all of our own code handling the deprecated functionality as invalid. And we would lure downstream projects into false security.

In this particular case, documenting the non-deprecated types only might be okay for the constructor. I agree that this actually helps downstream projects to detect the usage of deprecated features and prepare for the next major.

However, on the property and the return type of the getter method, I would consider this change to be harmful. We can demand that only `class-string`s are being passed to the constructor, but we cannot promise that `getFrom()` will always return a `class-string` as long as the class alias feature exists.

Reply:
Then `string` is still valid here. :point_up:

Reply:
Let's retarget to 3.0.x then @VincentLanglet ?

Reply:
Sure, it's done